fix: check before setting testing/testrun-status

AlthausKonstantin opened the pull request:
- the valid values of testing/testrun-status are defined on the server
- status_change changes the status only to allowed options
- otherwise status is not changed
- addresses https://github.com/RedHatQE/pylero/issues/168

AlthausKonstantin committed:
fix: check before setting testing/testrun-status

- the valid values of testing/testrun-status are defined on the server
- status_change changes the status only to allowed options
- otherwise status is not changed
- addresses https://github.com/RedHatQE/pylero/issues/168

AlthausKonstantin committed:
chore: run pre-commit

waynesun09 commented on src/pylero/test_run.py:
```diff
@@ -722,9 +723,12 @@ def _status_change(self):
+        if status != check_tr.status and status in valid_status_values:
             check_tr.status = status
-            check_tr.update()
+        check_tr.update()
```
With status not in valid_status_values, check_tr.update here also will update the finished_on status except condition in line 718 satisfied, is this expected?

AlthausKonstantin replied:
- yes, it is the original function of `_change_status`.
- IMHO, I think it would be cleaner to set `finished_on` in a new method. Then `_change_status` is doing what and only what the name implies.
- What are your thoughts- should I refactor this?

PS: Sorry for the late reply on my end.

waynesun09 replied:
The original function only do the update if the status has changed, else it will not update the finished_on value.

The finished_on and status is correlated in logic (reset to None when status is notrun or inprogress), so better to be updated together.

I don't think it need to be separated into a new function for the finished_on value.

Since there is no specific reason to update the finshed_on value, we don't need update line 731.

AlthausKonstantin replied:
Thanks for clarifying this for me. I will move the update into the if condition :)

AlthausKonstantin committed:
fix (test_run): when to update finished_on in _status_change

- resolves https://github.com/RedHatQE/pylero/pull/170/files/9dfc0d33278a9b3a4d3d2f035190f2dfea2766bf#r1496364770

AlthausKonstantin committed:
fix: correct rogue find and replace
